cmd/build/v2/services.go

```diff
@@ -14,115 +14,121 @@
 package v2
 
 import (
-	"context"
-	"fmt"
-	"strings"
-
-	"github.com/okteto/okteto/pkg/constants"
-	oktetoErrors "github.com/okteto/okteto/pkg/errors"
-	"github.com/okteto/okteto/pkg/format"
-	oktetoLog "github.com/okteto/okteto/pkg/log"
-	"github.com/okteto/okteto/pkg/model"
-	"github.com/okteto/okteto/pkg/okteto"
-	"golang.org/x/sync/errgroup"
+    "context"
+    "fmt"
+    "github.com/okteto/okteto/pkg/constants"
+    oktetoErrors "github.com/okteto/okteto/pkg/errors"
+    "github.com/okteto/okteto/pkg/format"
+    oktetoLog "github.com/okteto/okteto/pkg/log"
+    "github.com/okteto/okteto/pkg/model"
+    "github.com/okteto/okteto/pkg/okteto"
+    "golang.org/x/sync/errgroup"
+    "strings"
 )
 
 // GetServicesToBuild returns the services it has to build if they are not already built
 func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model.Manifest, svcToDeploy []string) ([]string, error) {
-	buildManifest := manifest.Build
-
-	if len(buildManifest) == 0 {
-		oktetoLog.Information("Build section is not defined in your okteto manifest")
-		return nil, nil
-	}
-
-	// check if images are at registry (global or dev) and set envs or send to build
-	toBuild := make(chan string, len(buildManifest))
-	g, _ := errgroup.WithContext(ctx)
-	for service := range buildManifest {
-
-		svc := service
-		g.Go(func() error {
-			return bc.checkServicesToBuild(svc, manifest, toBuild)
-		})
-	}
-
-	if err := g.Wait(); err != nil {
-		return nil, err
-	}
-	close(toBuild)
-
-	if len(toBuild) == 0 {
-		oktetoLog.Information("Images were already built. To rebuild your images run 'okteto build' or 'okteto deploy --build'")
-		if err := manifest.ExpandEnvVars(); err != nil {
-			return nil, err
-		}
-		return nil, nil
-	}
-
-	svcToDeployMap := map[string]bool{}
-	for _, svc := range svcToDeploy {
-		svcToDeployMap[svc] = true
-	}
-	svcsToBuildList := []string{}
-	for svc := range toBuild {
-		if _, ok := svcToDeployMap[svc]; len(svcToDeploy) > 0 && !ok {
-			continue
-		}
-		svcsToBuildList = append(svcsToBuildList, svc)
-	}
-	return svcsToBuildList, nil
+    buildManifest := manifest.Build
+
+    if len(buildManifest) == 0 {
+        oktetoLog.Information("Build section is not defined in your okteto manifest")
+        return nil, nil
+    }
+
+    // create a spinner to be loaded before checking if images needs to be built
+    oktetoLog.Spinner("Building images...")
+
+    // start the spinner
+    oktetoLog.StartSpinner()
+
+    // check if images are at registry (global or dev) and set envs or send to build
+    toBuild := make(chan string, len(buildManifest))
+    g, _ := errgroup.WithContext(ctx)
+    for service := range buildManifest {
+
+        svc := service
+        g.Go(func() error {
+            return bc.checkServicesToBuild(svc, manifest, toBuild)
+        })
+    }
+
+    if err := g.Wait(); err != nil {
+        return nil, err
+    }
+    close(toBuild)
+
+    if len(toBuild) == 0 {
+        defer oktetoLog.StopSpinner()
```

Comment:
Shouldn't we `defer` right after `StartSpinner` in order to be sure to stop the spinner in any case?